src/cell_group.hpp

```diff
@@ -231,7 +241,7 @@ class cell_group {
         auto handle = get_probe_handle(probe_id);
 
         auto sampler_index = uint32_t(samplers_.size());
-        samplers_.push_back({handle, s});
+        samplers_.push_back({handle, probe_id.gid, s});
```

Comment:
With this implementation the cell_group knows 'that the probe has a gid'.  tight coupling